### DISCUSSION TITLE
Global onSuccess callback in the MutationCache , interferes with local onSuccess used in useMutation hooks.

### DISCUSSION THREAD

**Discussion by emekaokoli** [TanStack/query#7890] [Q&A]
Hey there,

I've been using the global `mutationCache` for invalidating updates, and I've been relying on the `onSuccess` status to navigate away from the page when it's `true`. However, I've noticed that `onSuccess` always returns `false`, even when the mutation is successful. When I commented out the `mutationCache` section, I found that `onSuccess` starts working as intended, returning `true` or `false` based on the mutation outcome.

```
 mutationCache: new MutationCache({
 // Commenting it out returns true or false upon success or failure.
    // onSuccess: () => {
    //  queryClient.invalidateQueries()
    },
  }),
```

Now, I'm wondering if there's a way to use both the global invalidation and have `onSuccess` return the desired status of either `true` or `false`.

// where onSuccess is used.
```
 submitAction={() => {
     saveFormResponse({ id },
        { onSuccess: () => openChargeSheetModal() })
 }}
```

Also, here's the version I am using:
`"@tanstack/react-query": "^5.28.4",`

**Comment by TkDodo** [comment]
> When I commented out the mutationCache section, I found that onSuccess starts working as intended, returning true or false based on the mutation outcome.

that shouldn't have any influence. please show a minimal reproduction
